backend/de.metas.contracts/src/test/java/de/metas/contracts/commission/commissioninstance/services/margin/MarginCommissionConfigFactoryTest.java

```diff
@@ -0,0 +1,158 @@
+import de.metas.product.ProductId;
+import lombok.Builder;
+import org.adempiere.test.AdempiereTestHelper;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static io.github.jsonSnapshot.SnapshotMatcher.expect;
+import static io.github.jsonSnapshot.SnapshotMatcher.start;
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+public class MarginCommissionConfigFactoryTest
+{
+	private MarginCommissionConfigFactory marginCommissionConfigFactorySpy;
+
+	@BeforeAll
+	static void init()
+	{
+		start(AdempiereTestHelper.SNAPSHOT_CONFIG);
+		AdempiereTestHelper.get().init();
+	}
+
+	@BeforeEach
+	void beforeEach()
+	{
+		final CurrencyRepository currencyRepository = new CurrencyRepository();
+		final MoneyService moneyService = new MoneyService(currencyRepository);
+		final CustomerTradeMarginService customerTradeMarginService = new CustomerTradeMarginService(moneyService, new CustomerTradeMarginRepository());
+		marginCommissionConfigFactorySpy = Mockito.spy(new MarginCommissionConfigFactory(customerTradeMarginService));
+	}
+
+	@Test
+	public void givenValidRequest_whenCreateForNewCommissionInstances_thenReturnOneCommissionConfig()
+	{
+		//given
+		final OrgId orgId = AdempiereTestHelper.createOrgWithTimeZone();
+		final BPartnerId salesRepId = BPartnerId.ofRepoId(1);
+		final BPartnerId customerBPartnerId = BPartnerId.ofRepoId(2);
+		final LocalDate commissionDate = LocalDate.of(2021, 9, 13);
+		final ProductId transactionProductId = ProductId.ofRepoId(104);
+		final ProductId commissionProductId = ProductId.ofRepoId(103);
+
+		final I_C_Flatrate_Term mediatedContract = contractAndComplementaryRecordsBuilder()
+				.commissionProductId(commissionProductId)
+				.salesRepId(salesRepId)
+				.orgId(orgId)
+				.build();
+
+		final CommissionConfigProvider.ConfigRequestForNewInstance requestForNewInstance = CommissionConfigProvider.ConfigRequestForNewInstance.builder()
+				.commissionDate(commissionDate)
+				.commissionHierarchy(Hierarchy.EMPTY_HIERARCHY)
+				.customerBPartnerId(customerBPartnerId)
+				.salesRepBPartnerId(salesRepId)
+				.orgId(orgId)
+				.salesProductId(transactionProductId)
+				.commissionTriggerType(CommissionTriggerType.InvoiceCandidate)
+				.build();
+
+		Mockito.doReturn(ImmutableList.of(mediatedContract))
+				.when(marginCommissionConfigFactorySpy)
+				.retrieveContracts(salesRepId, orgId, commissionDate);
+		//when
+		final List<CommissionConfig> configs = marginCommissionConfigFactorySpy.createForNewCommissionInstances(requestForNewInstance);
+
+		//then
+		expect(configs).toMatchSnapshot();
+	}
+
+	@Builder(builderMethodName = "contractAndComplementaryRecordsBuilder")
+	private I_C_Flatrate_Term createContractAndComplementaryRecords(
```

Comment:
Please extract/refactor this and place it as a top-level-class next to `TestCommissionContract`...probably, rename TestCommissionContract to TestHierarchyCommissionContract..
Might be usefull for upcoming unit and cucumber-tests

PS: of if you think the effort is not worth it, pls let me know